[Program: GCI] Replaced edit in popup to direct edit on ProfileFragment

bartekpacia opened the pull request:
### Description
So, after some fighting, I managed to do this.

* In order to achieve nice scrolling effect (`CollapsingToolbarLayout`), I had to include separate Toolbar in each Fragment. It creates some code duplication, but the advantage is that we'll have more flexibility in the future.

* I didn't create separate `TextView`s for every input, instead I simple change `enabled` state of `TextInputEditText`.

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors).


### How Has This Been Tested?
I've tested it manually. Demo:
![demo](https://user-images.githubusercontent.com/40357511/82670351-430bb400-9c3d-11ea-8a83-87db4c68a09d.gif)

### Checklist:


- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas



**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

- [x] New and existing unit tests pass locally with my changes

bartekpacia committed:
feat: Replaced edit in popup to edit in place on profile screen

* added onOptionsItemSelected() default implementation to BaseFragment

annabauza commented on app/src/main/java/org/systers/mentorship/view/activities/MainActivity.kt:
```diff
@@ -70,19 +70,19 @@ class MainActivity: BaseActivity() {
-                }
-                else -> false
-            }
+//    override fun onCreateOptionsMenu(menu: Menu): Boolean {
```
Please remove redundant code.

bartekpacia replied:
Done

annabauza commented on app/src/main/java/org/systers/mentorship/view/fragments/ProfileFragment.kt:
```diff
@@ -37,34 +54,149 @@ class ProfileFragment : BaseFragment() {
+            }
+
+            switches.forEach {
+                it.isEnabled = editing
```
I think it will be great to deeper your knowledge about data binding :) not everything we have to do in code. Android is a great place where we have many concept involved and we can get same effect with in many different coding principals and patterns. Great work!

bartekpacia replied:
Yeah, you're right. This can be done better way using DataBinding. I'm gonna fix it.

annabauza replied:
Fantastic!

annabauza reviewed (changes requested):
please remove redundant code!

bartekpacia commented:
Okay, I've implemented management of `enabled` state and `background` of those EditTexts using DataBinding. It turned out to be really cool, I have always underestimated it :)

bartekpacia commented:
Well, I don't know when but this PR got quite massive. I'll try to perform better in the future😅

 I'm open to advices and critisism.

annabauza commented on app/src/main/java/org/systers/mentorship/view/fragments/ProfileFragment.kt:
```diff
@@ -20,7 +36,25 @@ class ProfileFragment : BaseFragment() {
+         * Binding adapter used for setting the right background for EditTexts in this fragment's layout.
+         */
+        @JvmStatic
+        @BindingAdapter("android:background")
```
Can't see where you are using it in layouts

bartekpacia replied:
It's automatically used in `TextInputEditText`s, when I set `android:background="${editing}"`

annabauza replied:
Found it 👍

annabauza reviewed (approved):
Thank you @bartekpacia this is exactly what I expected to see. Well done, you have shown good understanding of mvvm and data binding 👍

bartekpacia commented:
Thanks for guiding me!

ilyamaier reviewed (commented):
`fragmentProfileBinding.user = profileViewModel.user`
@bartekpacia have you tested it? I was also doing this task and I needed to implement deep copy in order to achieve that.

bartekpacia commented:
@IlyaMaier I've tested it now (again), everything works fine.

devansh-299 commented:
@bartekpacia can please attach any gif etc. to show the changes you made. Your link is not working

bartekpacia commented:
Sorry for that, it was in the GCI folder on my Google Drive but after GCI I deleted it.
Updated with GIF.
Structure of the app's changed a lot since that time and there's many conflicts, so I'd suggest closing this.

devansh-299 commented:
@bartekpacia nice work! You can rework on this later if you want. For now I am closing this PR
